### PR TITLE
Added detail connecting k8s cluster access to user roles.

### PR DESCRIPTION
**Pull request by stevenGravy**
Users were unable to understand how to connect the cluster and how to access with their users' role. Added detail on roles, using user traits and autopopulating from SSO.   Note that our db access guides have these steps included.

### CODE REVIEW

**Comment by stevenGravy** [comment]
Addresses #10480

File: docs/pages/kubernetes-access/getting-started/agent.mdx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ---
-title: Connect a Kubernetes Cluster to Teleport
-description: Connecting a Kubernetes cluster to Teleport
+title: Connect Kubernetes Cluster to Teleport
+description: Connecting Kubernetes cluster to Teleport
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/agent.mdx:3]: Why remove `a`? I think it read better before.

File: docs/pages/kubernetes-access/getting-started/agent.mdx [hunk 2]
```diff
@@ -64,14 +64,118 @@ $ helm install teleport-agent teleport/teleport-kube-agent --set kubeClusterName
   --set proxyAddr=${PROXY?} --set authToken=${TOKEN?} --create-namespace --namespace=teleport-agent
 ```
 
+## Step 3/4 Configure User and Roles for Access
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/agent.mdx:67]: I don't think we're doing title-case for section headers anymore. @ptgott?

File: docs/pages/kubernetes-access/getting-started/agent.mdx [hunk 3]
```diff
@@ -64,14 +64,118 @@ $ helm install teleport-agent teleport/teleport-kube-agent --set kubeClusterName
...
+
+The `kubernetes_groups`, `kubernetes_labels` and `kubernetes_users` determine the access from a Teleport
+role to the Kubernetes Clusters.
+
+```YAML
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/agent.mdx:72]:
```suggestion
```yaml
```

File: docs/pages/kubernetes-access/getting-started/agent.mdx [hunk 4]
```diff
@@ -64,14 +64,118 @@ $ helm install teleport-agent teleport/teleport-kube-agent --set kubeClusterName
...
+  # Sets which groups in the Kubernetes the user should access as.  The group must
+  # already exist in the Kubernetes cluster. Example: system:masters
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/agent.mdx:74]: The indentation looks off here and below.

File: docs/pages/kubernetes-access/getting-started/agent.mdx [hunk 5]
```diff
@@ -64,14 +64,118 @@ $ helm install teleport-agent teleport/teleport-kube-agent --set kubeClusterName
...
+    - '{{internal.kubernetes_users}}'
+```
+
+<Admonition type="tip" title="Group or User required">
+At least one user or group available in the Kubernetes cluster must be set in a user's Teleport role to access the Kubernetes cluster.
+</Admonition>
+
+<Tabs>
+  <TabItem label="Specify access in Role">
+To add a group like `system:masters` into the access  role follow these steps.  Get
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/agent.mdx:92]:
```suggestion
To add a group like `system:masters` into the access role follow these steps.  Get
```

Remove extra whitespace

File: docs/pages/kubernetes-access/getting-started/agent.mdx [hunk 6]
```diff
@@ -64,14 +64,118 @@ $ helm install teleport-agent teleport/teleport-kube-agent --set kubeClusterName
...
+  ```yaml
+  traits:
+    kubernetes_users:
+    - alice@mycompany1.com
+    kubernetes_groups:
+    - system:masters
+  ```
+
+  ```code
+  tctl create -f alice.yaml
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/agent.mdx:142]:
```suggestion
  $ tctl create -f alice.yaml
```

File: docs/pages/kubernetes-access/getting-started/agent.mdx (hunk 1 above)
Comment (stevenGravy) [reply docs/pages/kubernetes-access/getting-started/agent.mdx]: restored, this got mixed up in the copying.

File: docs/pages/kubernetes-access/getting-started/agent.mdx [hunk 7]
```diff
@@ -64,14 +64,118 @@ $ helm install teleport-agent teleport/teleport-kube-agent --set kubeClusterName
   --set proxyAddr=${PROXY?} --set authToken=${TOKEN?} --create-namespace --namespace=teleport-agent
 ```
 
+## Step 3/4 Configure User and Roles for Access
+
+The `kubernetes_groups`, `kubernetes_labels` and `kubernetes_users` determine the access from a Teleport
+role to the Kubernetes Clusters.
```
Comment (ptgott) [inline docs/pages/kubernetes-access/getting-started/agent.mdx:70]:
```suggestion
role to a Kubernetes cluster.
```

**Review by ptgott — COMMENTED** [review]
Leaving a partial round of comments that includes everything up to the beginning of the Tabs component. I will need to finish my review, but feel free to make changes.

**Review by ptgott — COMMENTED** [review]
@stevenGravy Let me know when you've responded to Zac and my comments, and I'll give this another look!

**Comment by r0mant** [comment]
@stevenGravy Are you still planning to apply suggestions here?

**Comment by stevenGravy** [comment]
Already done separately

**Pull request closed without merging**